### PR TITLE
Update submission count on change, enable CI for new tests

### LINKED ISSUE #1123: Add submission count to LOI when writing submissions

### LINKED ISSUE #1126: Set up separate project for CI testing
We currently use the same project for dev and testing.

@os-micmec FYI for future work

### PR DESCRIPTION
**Pull request by gino-m**
- [x] Introduced offline unit tests using Jasmine. Tried several options, incl. in-memory Firebase impls and testing libs, but deps either didn't work as advertised, or were not actively maintained.
- [x] Created minimal shared test helpers for working with Firestore.
- [x] Renamed build dir from `lib` to `dist` for consistency w/web package
- [x] Replaced `mocha` test runner with `jasmine` for consistency w/web
- [x] Adds a job to run GitHub Actions on PRs

Fixes #1123
Fixes #1126

image

### CODE REVIEW

File: functions/test/on-write-submission.test.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import { onWriteSubmission } from '../src/index'; // relative path to functions code
+
+
+// describe("onWriteSubmissionTest", async () => {
+  it("should update count on new submission", async () => {
+    const before = test.firestore.makeDocumentSnapshot({}, 'survey/100/submissions/123');
+    const after = test.firestore.makeDocumentSnapshot({loiId: '999'}, 'survey/100/submissions/123');
+    const change = test.makeChange(before, after);
+    const wrapped = test.wrap(onWriteSubmission);
+    await wrapped(change);
```
Comment (gino-m) [inline functions/test/on-write-submission.test.ts:38]: @os-micmec This line is timing out

**Comment by gino-m** [comment]
@os-micmec @DaoyuT  Making some progress on testing using stubs with Jasmine. I moved from Mocha to Jasmine and also reorganized our test and output folder structure and naming to bring Cloud Functions impls close to web/. Still WIP, I'll ping once tests are ready.

**Comment by gino-m** [comment]
@os-micmec PTAL?

**Review by DaoyuT — APPROVED** [review]
LGTM

File: functions/src/on-write-submission.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ */
+
+// import * as admin from 'firebase-admin';
+import {Change, EventContext} from 'firebase-functions';
+import {DocumentSnapshot} from 'firebase-functions/v1/firestore';
+import { surveyPathTemplate } from '@/on-write-survey';
+import {db} from '@/common/context';
+
+// Template for submission write triggers capturing survey and submission ids.
+export const submissionPathTemplate = `${surveyPathTemplate}/submissions/{submissionId}`;
```
[the remark this answers is not in the document]
Comment (gino-m) [reply functions/src/on-write-submission.ts:25]: It actually shouldn't have been in `onWriteSurvey`; I was trying to place it close to the code that actually depends on those template variables. I've done some refactoring and moved this out, sending commit now.

**Comment by gino-m** [comment]
@DaoyuT Did some cleanup, RFAL

**Review by DaoyuT — APPROVED** [review]

**Pull request merged**